### ISSUE TITLE
`mixed_attributes_style` should allow mixing outer cfgs and inner doccomments

### ISSUE THREAD

**Issue by apoelstra** [rust-lang/rust-clippy#12435]
### Summary

The `mixed_attributes_style` treats doccomments as attributes, which I suppose they technically are, but this is not how users perceive them. It is common with modules to use inner doccomments (and inner style lints, clippy allows, etc., for that matter, but ok, I can accept that the lint doesn't like that) while having outer `cfg` gates for features.

### Reproducer

I tried this code:

```rust
#[cfg(test)]
mod tests {
    //! Doccomment for my tests module
}
```

This triggers the lint:

```
warning: item has both inner and outer attributes
 --> src/main.rs:2:1
  |
2 | / #[cfg(test)]
3 | | mod tests {
4 | |     //! Doccomment for my tests module
  | |______________________________________^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#mixed_attributes_style
  = note: `#[warn(clippy::mixed_attributes_style)]` on by default

warning: `clippy-demo` (bin "clippy-demo") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 6.30s
```

But I think either fix would be more confusing to read (either add a `#![cfg(test)]` which I have literally never seen in Rust code, or move the doccomment out, which makes it much harder for readers of the code to see that this is a module-level "header comment").

### Version

```text
rustc 1.73.0 (cc66ad468 2023-10-03)
binary: rustc
commit-hash: cc66ad468955717ab92600c770da8c1601a4ff33
commit-date: 2023-10-03
host: x86_64-unknown-linux-gnu
release: 1.73.0
LLVM version: 17.0.2
```

**Comment by nyurik** [comment]
I also noticed this lint when running `cargo +nightly clippy -p martin-tile-utils` on https://github.com/maplibre/martin (observed in `b92a407`):

```
warning: item has both inner and outer attributes
   --> martin-tile-utils/src/lib.rs:281:1
    |
281 | / #[cfg(test)]
282 | | mod tests {
283 | |     #![allow(clippy::unreadable_literal)]
    | |_________________________________________^
```
